README.md

```diff
@@ -257,6 +216,44 @@ try {
 
 See [API documentation](docs/api.md) for more example and full API reference information.
 
+## Generator version vs Template version
+
+The generator is a tool that you can use to generate whatever you want, taking AsyncAPI specification file as an input. A template is a tool that uses Generator features and helpers to specify what should be generated.
+
+In other words, a template depends on the generator and its features. For example it might work with the latest version of the generator but not the previous ones. 
+
+Owner of the template specifies in the configuration what version of the generator is the template compatible with:
+```bash
+"generator": ">=0.50.0 <2.0.0",
+```
+
+The generator doesn't work in case template is not compatible:
+```bash
+Something went wrong:
+Error: This template is not compatible with the current version of the generator (0.50.0). This template is compatible with the following version range: >=0.60.0 <2.0.0.
+    at Generator.validateTemplateConfig (/Users/wookiee/.nvm/versions/node/v12.16.1/lib/node_modules/@asyncapi/generator/lib/generator.js:678:13)
+    at Generator.loadTemplateConfig (/Users/wookiee/.nvm/versions/node/v12.16.1/lib/node_modules/@asyncapi/generator/lib/generator.js:663:16)
+    at Generator.generate (/Users/wookiee/.nvm/versions/node/v12.16.1/lib/node_modules/@asyncapi/generator/lib/generator.js:146:18)
+    at processTicksAndRejections (internal/process/task_queues.js:97:5)
+    at async /Users/wookiee/.nvm/versions/node/v12.16.1/lib/node_modules/@asyncapi/generator/cli.js:135:7
+```
+
+In case you use generator CLI and a specific template on production, it is safer to lock to a specific version of the template and the generator. 
```

Comment:
```suggestion
In case you use Generator CLI and a specific template on production, it is safer to lock to a specific version of the template and the Generator. 
```